src/k8s/pkg/k8sd/api/certificates_refresh.go

```diff
@@ -154,7 +154,12 @@ func refreshCertsRunWorker(s state.State, r *http.Request, snap snap.Snap) respo
 				return fmt.Errorf("failed to generate CSR for %s: %w", csr.name, err)
 			}
 
-			pubKey, err := pkiutil.LoadRSAPublicKey(clusterConfig.Certificates.GetK8sdPublicKey())
+			k8sdPublicKey := clusterConfig.Certificates.GetK8sdPublicKey()
+			if k8sdPublicKey == "" {
+				return fmt.Errorf("k8sd public key not set")
+			}
+
+			pubKey, err := pkiutil.LoadRSAPublicKey(k8sdPublicKey)
```

Comment:
NOTE on style of `k8sdPublicKey` and `pubKey`: Naming wise, they appear to be entities of the same thing, but one of them is the key, the other is a PEM encoded string. 

i'd recommend having the variable name _describe_ what it contains. try `publicKeyPEM` and `publicKey` instead. That makes it easier to follow the rest of the code.